network/common/src/test/java/org/apache/spark/network/buffer/LargeByteBufferHelperSuite.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.network.buffer;
+
+import java.io.*;
+import java.nio.channels.FileChannel;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class LargeByteBufferHelperSuite {
+
+  @Test
+  public void testMapFile() throws IOException {
+    File testFile = File.createTempFile("large-byte-buffer-test", ".bin");
+    try {
+      testFile.deleteOnExit();
+      OutputStream out = new FileOutputStream(testFile);
+      byte[] buffer = new byte[1 << 16];
+      long len = ((long)buffer.length) + Integer.MAX_VALUE + 1;
```

Comment:
actually, the jvm **cannot** mmap a region of a file larger than 2GB, precisely because `channel.map` returns a `ByteBuffer` -- so the code actually has to [break it into a chunks](https://github.com/squito/spark/blob/112c49e7074dc3a53f7425f6b93fbff7608f6a23/network/common/src/main/java/org/apache/spark/network/buffer/LargeByteBufferHelper.java#L80).  (maybe that is what you meant, just wanted to clarify.)

Though honestly, that isn't even the main thing I wanted to test. really I just wanted to test the dumb basic stuff, eg. `position` isn't accidentally treated as an int somewhere so it overflows.  I know I'm really prone to messing up dumb stuff like that if I don't test it (and even if it is correct now, to prevent future regressions).